packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -485,26 +488,55 @@ export class RpcService implements AbstractRpcService {
     } catch (error) {
       if (error instanceof HttpError) {
         const status = error.httpStatus;
-        if (status === 405) {
-          throw rpcErrors.methodNotFound();
+        if (status === 401) {
+          throw new JsonRpcError(-33100, 'Unauthorized.', {
+            httpStatus: status,
+          });
+        }
+        if (status === 402 || status === 404) {
+          throw rpcErrors.resourceUnavailable({
+            message: `RPC endpoint server error (HTTP ${status})`,
+            data: {
+              httpStatus: status,
+            },
+          });
+        }
+        if (status === 405 || status === 501) {
+          throw rpcErrors.methodNotFound({
+            data: {
+              httpStatus: status,
+            },
+          });
         }
         if (status === 429) {
           throw rpcErrors.limitExceeded({
             message: 'Request is being rate limited.',
+            data: {
+              httpStatus: status,
+            },
           });
         }
-        if (status === 503 || status === 504) {
-          throw rpcErrors.internal({
-            message:
-              'Gateway timeout. The request took too long to process. This can happen when querying logs over too wide a block range.',
+        if (status >= 500 && status < 600) {
```

Comment:
Nit: Statuses can't be above 5xx, so we could simplify this:

``` suggestion
        if (status >= 500) {

Reply:
Good point. If the server somehow is returning an error >=600, it seems fair to call that "resource unavailable" anyway (that would be a misconfigured server rather than an internal error). So this suggestion is good even we somehow hit this case.